sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/basicOperators.scala

```diff
@@ -518,8 +543,8 @@ private[sql] object Expand {
 case class Expand(
     projections: Seq[Seq[Expression]],
     output: Seq[Attribute],
-    child: LogicalPlan) extends UnaryNode {
-
+    child: LogicalPlan,
+    constraintsBase: Seq[Expression]) extends UnaryNode {
```

Comment:
This parameter is not documented.

Also, this operator is kind of violating the style of logical operators.  The goal here is for query nodes to be correct by construction (i.e. figure out things like constraints by inspecting themselves and their children).

It would be great if we could move constraint pruning logic into `Expand`.